code/game/machinery/computer/dropship_weapons.dm

```diff
@@ -1,3 +1,116 @@
+
+/datum/camera_manager
+	var/map_name
+	var/datum/shape/rectangle/current_area
+	var/atom/movable/screen/map_view/cam_screen
+	var/atom/movable/screen/background/cam_background
+	var/list/range_turfs = list()
+	/// The turf where the camera was last updated.
+	var/turf/last_camera_turf
+	var/target_x
+	var/target_y
+	var/target_z
+	var/list/cam_plane_masters
+
+/datum/camera_manager/New(map_ref)
+	map_name = map_ref
+	cam_screen = new
+	cam_screen.name = "screen"
+	cam_screen.assigned_map = map_name
+	cam_screen.del_on_map_removal = FALSE
+	cam_screen.screen_loc = "[map_name]:1,1"
+	cam_background = new
+	cam_background.assigned_map = map_name
+	cam_background.del_on_map_removal = FALSE
+
+	cam_plane_masters = list()
+	for(var/plane in subtypesof(/atom/movable/screen/plane_master) - /atom/movable/screen/plane_master/blackness)
+		var/atom/movable/screen/plane_master/instance = new plane()
+		instance.assigned_map = map_name
+		instance.del_on_map_removal = FALSE
+		if(instance.blend_mode_override)
+			instance.blend_mode = instance.blend_mode_override
+		instance.screen_loc = "[map_name]:CENTER"
+		cam_plane_masters += instance
+
+/datum/camera_manager/Destroy(force, ...)
+	. = ..()
+	range_turfs = null
+	QDEL_NULL(cam_background)
+	QDEL_NULL(cam_screen)
+	QDEL_NULL(cam_plane_masters)
+
+/datum/camera_manager/proc/register(mob/user)
+	user.client.register_map_obj(cam_background)
+	user.client.register_map_obj(cam_screen)
+	for(var/plane in cam_plane_masters)
+		user.client.register_map_obj(plane)
+
+/datum/camera_manager/proc/clear_camera()
+	current_area = null
+	target_x = null
+	target_y = null
+	target_z = null
+	update_active_camera()
+
+/datum/camera_manager/proc/set_camera_obj(obj/target, w, h)
+	set_camera_rect(target.x, target.y, target.z, w, h)
+
+/datum/camera_manager/proc/set_camera_turf(turf/target, w, h)
+	set_camera_rect(target.x, target.y, target.z, w, h)
```

Comment:
why have both separate? both are /atom and it seems like it'd work for this purpose

Reply:
You're right!